cpp/oneapi/dal/backend/primitives/objective_function/logloss_functors_dpc.cpp

```diff
@@ -0,0 +1,530 @@
+        set_csr_data(q_, *sp_handle_, static_cast<const csr_table&>(data_));
+    }
+}
+
+template <typename Float>
+logloss_hessian_product<Float>::logloss_hessian_product(sycl::queue& q,
+                                                        const table& data,
+                                                        Float L2,
+                                                        bool fit_intercept,
+                                                        std::int64_t bsz)
+        : q_(q),
+          data_(data),
+          L2_(L2),
+          fit_intercept_(fit_intercept),
+          n_(data.get_row_count()),
+          p_(data.get_column_count()),
+          bsz_(bsz == -1 ? get_block_size(n_, p_) : bsz) {
+    this->reserve_memory();
+}
+
+template <typename Float>
+logloss_hessian_product<Float>::logloss_hessian_product(sycl::queue& q,
+                                                        comm_t comm,
+                                                        const table& data,
+                                                        Float L2,
+                                                        bool fit_intercept,
+                                                        std::int64_t bsz)
+        : q_(q),
+          comm_(comm),
+          data_(data),
+          L2_(L2),
+          fit_intercept_(fit_intercept),
+          n_(data.get_row_count()),
+          p_(data.get_column_count()),
+          bsz_(bsz == -1 ? get_block_size(n_, p_) : bsz) {
+    this->reserve_memory();
+}
+
+template <typename Float>
+ndview<Float, 1>& logloss_hessian_product<Float>::get_raw_hessian() {
+    return raw_hessian_;
+}
+
+template <typename Float>
+sycl::event logloss_hessian_product<Float>::compute_with_fit_intercept(const ndview<Float, 1>& vec,
+                                                                       ndview<Float, 1>& out,
+                                                                       const event_vector& deps) {
+    ONEDAL_PROFILER_TASK(compute_hessp_with_fit_intercept, q_);
+    auto* const tmp_ptr = tmp_gpu_.get_mutable_data();
+    ONEDAL_ASSERT(vec.get_dimension(0) == p_ + 1);
+    ONEDAL_ASSERT(out.get_dimension(0) == p_ + 1);
+    auto fill_buffer_event = fill<Float>(q_, buffer_, Float(1), deps);
+    auto out_suf = out.get_slice(1, p_ + 1);
+    auto tmp_suf = tmp_gpu_.slice(1, p_);
+    auto out_bias = out.get_slice(0, 1);
+    auto vec_suf = vec.get_slice(1, p_ + 1);
+    ndview<Float, 1> tmp_ndview = tmp_gpu_;
+
+    sycl::event fill_out_event = fill<Float>(q_, out, Float(0), deps);
+
+    const Float v0 = vec.at_device(q_, 0, deps);
+    event_vector last_iter_deps = { fill_buffer_event, fill_out_event };
+
+    if (data_.get_kind() == dal::csr_table::kind()) {
+        const auto* const hess_ptr = raw_hessian_.get_data();
+        auto* const out_ptr = out.get_mutable_data();
+        auto* const buffer_ptr = buffer_.get_mutable_data();
+
+        sycl::event event_xv = gemv(q_,
```

Comment:
its better to separate this gemv with { } and add kernel profilers(just a suggestion)

Reply:
I believe that profiler should be added inside gemv function in sparse_blas kernel